ankaios_sdk/_components/complete_state.py

```diff
@@ -0,0 +1,320 @@
+            self._complete_state.desiredState.configs.configs[key].CopyFrom(
+                _to_config_item(value)
+            )
+
+    def get_configs(self) -> dict:
+        """
+        Gets the configurations from the complete state.
+
+        Returns:
+            dict: The configurations from the complete state
+        """
+        return self._configs
+
+    @staticmethod
+    def from_manifest(manifest: Manifest) -> 'CompleteState':
+        """
+        Creates a CompleteState instance from a Manifest.
+
+        Args:
+            manifest (Manifest): The manifest to create the
+                complete state from.
+        """
+        state = CompleteState()
+        state._complete_state = _ank_base.CompleteState()
+        dict_state = manifest._manifest
+        state._set_api_version(
+            dict_state.get("apiVersion", state.get_api_version())
+        )
+        state._workloads = []
+        if dict_state.get("workloads") is not None:
+            for workload_name, workload_dict in \
+                    dict_state.get("workloads").items():
+                state._workloads.append(
+                    Workload._from_dict(workload_name, workload_dict)
+                )
+        if dict_state.get("configs") is not None:
+            state.set_configs(dict_state.get("configs"))
+        return state
+
+    def to_dict(self) -> dict:
+        """
+        Returns the CompleteState as a dictionary
+
+        Returns:
+            dict: The CompleteState as a dictionary.
+        """
+        data = {
+            "desired_state": {
+                "api_version": self.get_api_version(),
+                "workloads": {},
+                "configs": self._configs
+            },
+            "workload_states": {},
+            "agents": {}
+        }
+        for wl in self._workloads:
+            data["desired_state"]["workloads"][wl.name] = \
+                wl.to_dict()
+        wl_states = self._workload_state_collection.get_as_dict()
+        for agent_name, exec_states in wl_states.items():
+            data["workload_states"][agent_name] = {}
+            for workload_name, exec_states_id in exec_states.items():
+                data["workload_states"][agent_name][workload_name] = {}
+                for workload_id, exec_state in exec_states_id.items():
+                    data["workload_states"][agent_name][workload_name][
+                        workload_id] = exec_state.to_dict()
+        data["agents"] = self.get_agents()
+        return data
+
+    def _to_proto(self) -> _ank_base.CompleteState:
+        """
+        Returns the CompleteState as a proto message.
+
+        Returns:
+            _ank_base.CompleteState: The protobuf message representing
+                the complete state.
+        """
+        for workload in self._workloads:
+            self._complete_state.desiredState.workloads.\
+                workloads[workload.name].CopyFrom(workload._to_proto())
```

Comment:
This should not be needed after the fix on line [236](https://github.com/eclipse-ankaios/ank-sdk-python/pull/1/files#r1829070391):
```suggestion
```

Reply:
Done.